[32mINFO    [0m pytest_operator.plugin:plugin.py:862 Model status:

Model  Controller           Cloud/Region         Version  SLA          Timestamp
test   localhost-localhost  localhost/localhost  3.4.4    unsupported  01:59:33Z

App                  Version  Status  Scale  Charm                Channel      Rev  Exposed  Message
postgresql           14.12    active      3  postgresql                          0  no       
postgresql-test-app           active      1  postgresql-test-app  latest/edge  194  no       received database credentials of the first database

Unit                    Workload     Agent      Machine  Public address  Ports     Message
postgresql-test-app/0*  active       idle       3        10.204.68.146             received database credentials of the first database
postgresql/1            active       executing  1        10.204.68.114   5432/tcp  
postgresql/2*           active       idle       2        10.204.68.220   5432/tcp  Primary
postgresql/3            maintenance  executing  4        10.204.68.56    5432/tcp  reinitialising replica

Machine  State    Address        Inst id        Base          AZ  Message
1        started  10.204.68.114  juju-fe28fc-1  ubuntu@22.04      Running
2        started  10.204.68.220  juju-fe28fc-2  ubuntu@22.04      Running
3        started  10.204.68.146  juju-fe28fc-3  ubuntu@22.04      Running
4        started  10.204.68.56   juju-fe28fc-4  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:868 Juju error logs:

machine-2: 01:32:34 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:32:34 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-2: 01:32:34 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-1: 01:32:35 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:32:35 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-1: 01:32:36 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-0: 01:32:36 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:32:36 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-0: 01:32:37 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:32:37 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:32:37 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-test-app-0: 01:32:39 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-4: 01:42:15 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-4: 01:42:15 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-3: 01:42:16 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-postgresql-2: 01:47:59 ERROR unit.postgresql/2.juju-log Failed to list PostgreSQL database users: connection to server at "10.204.68.56", port 5432 failed: timeout expired

unit-postgresql-2: 01:48:09 ERROR unit.postgresql/2.juju-log Failed to list PostgreSQL database users: connection to server at "10.204.68.56", port 5432 failed: timeout expired

unit-postgresql-test-app-0: 01:51:55 ERROR unit.postgresql-test-app/0.juju-log first-database:4: Unable to read result
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 446, in _stop_continuous_writes
    with open(LAST_WRITTEN_FILE, "r") as fd:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/last_written_value'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 444, in _stop_continuous_writes
    for attempt in Retrying(stop=stop_after_delay(60), wait=wait_fixed(5)):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/tenacity/__init__.py", line 443, in __iter__
    do = self.iter(retry_state=retry_state)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/tenacity/__init__.py", line 376, in iter
    result = action(retry_state)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/tenacity/__init__.py", line 419, in exc_check
    raise retry_exc from fut.exception()
tenacity.RetryError: RetryError[<Future at 0x7efeaaf6f010 state=finished raised FileNotFoundError>]
unit-postgresql-2: 01:52:22 ERROR unit.postgresql/2.juju-log Failed to list PostgreSQL database users: connection to server at "10.204.68.114", port 5432 failed: FATAL:  the database system is shutting down


[32mINFO    [0m pytest_operator.plugin:plugin.py:947 Forgetting model main...